### DISCUSSION TITLE
Fonts clarification

### DISCUSSION THREAD

**Discussion by tquest1** [nextcloud/all-in-one#1643] [Questions] [answered]
I've read the previously answered (How to add fonts to Collabora container #870) re: font volume implementation that has been added to the latest release, but I wanted to know how do we actually go about installing fonts.

I have Nextcloud AIO v4.0.1 installed via a docker-compose stack in Portainer.

I have copied some truetype fonts to the nextcloud_aio_collabora_fonts volume by copying them to the path:
/var/lib/docker/volumes/nextcloud_aio_collabora_fonts/_data .

I can confirm that they are visible inside the nextcloud-aio-collabora container at: /opt/cool/systemplate/tmpfonts , but I don't see the fonts in the office gui. 

Are there any other steps required? Also - what is the correct way to remove some of the pre-installed fonts - I'd like to remove the unused 'Noto' font languages as the font drop down is too long with all of the various noto fonts.

**Comment by szaimen** [comment]
Hi, usually you should be able to upload additional fonts easily from the office admin settings, so no manual copying of the files on the server should be needed. However the last time I tried to upload font files it produced an error which should probably get reported to https://github.com/nextcloud/richdocuments/issues if you can reproduce it.

> I'd like to remove the unused 'Noto' font languages as the font drop down is too long with all of the various noto fonts.

This is currently not possible afaik and would be a feature request for either the richdocuments app or the collabora software.

**Reply by tquest1** [reply]
When uploading files from the Office admin settings - there is no error message for me & the font appears under Available fonts but is not actually available in the app. I had a similar problem with a non-AIO standalone Collabora server & after following the Office settings instructions on the settings page (see below) to add the mentioned URL to the coolwsd.xml everything worked. 

`Make sure to set this URL: https://my.server.net/apps/richdocuments/settings/fonts.json in the coolwsd.xml file of your Collabora Online server to ensure the added fonts get loaded automatically.`

`<remote_font_config>
	<url desc="URL of optional JSON file that lists fonts to be included in Online" type="string" default="">https://my.server.net/apps/richdocuments/settings/fonts.json</url>
</remote_font_config>`

How can this be done with AIO included Collabora?

**Reply by szaimen** [reply]
First of all, I still cannot update any font to try out if it works. It just always shows an error: `Font format not supported`. Can you maybe share a for you working file with me that I could use for testing? Thanks a lot!

Second, even if I set `--o:remote_font_config.url=https://mydomain.com/apps/richdocuments/settings/fonts.json` via command line as additional value of the docker container, it still shows the warning for me in the office settings. Do you know if this is expected?

**Comment by tquest1** [comment]
Basically any truetype font that I try works e.g. the attached file.
[arial.ttf.zip](https://github.com/nextcloud/all-in-one/files/10344312/arial.ttf.zip)

![extra-fonts](https://user-images.githubusercontent.com/43093959/210561842-7742f4de-444c-4da5-8a4f-ffd94789c9d7.png)

> Second, even if I set `--o:remote_font_config.url=https://mydomain.com/apps/richdocuments/settings/fonts.json` via command line as additional value of the docker container, it still shows the warning for me in the office settings. Do you know if this is expected?

How do you actually edit the above value so that I can try it?

**Reply by szaimen** [reply]
> Basically any truetype font that I try works e.g. the attached file.
> [arial.ttf.zip](https://github.com/nextcloud/all-in-one/files/10344312/arial.ttf.zip)

thanks. so yeah it doesnt work with that too. No idea why. I always get 
![image](https://user-images.githubusercontent.com/42591237/210562820-56163de1-f3c9-4738-aefe-826a187c14bd.png)

**Reply by szaimen** [reply]
> How do you actually edit the above value so that I can try it?

I've created this PR https://github.com/nextcloud/all-in-one/pull/1663 which will fix it with the next version. I'll let you know once it is available in the beta channel.

**Comment by szaimen** [comment, marked as answer]
This is now released with v4.1.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Reply by tquest1** [reply]
I can confirm that it is now working with the latest [v4.1.0 Beta](https://github.com/nextcloud/all-in-one/releases/tag/v4.1.0) - thanks a lot for your rapid fix!

![fonts2](https://user-images.githubusercontent.com/43093959/210837552-e1bf7144-f0c1-427f-b1cf-29563d5b374b.png)

<img width="377" alt="fonts1" src="https://user-images.githubusercontent.com/43093959/210837605-00307a6e-5b14-4df6-88bc-7b3cd7e17800.png">

**Reply by szaimen** [reply]
Great to see! Now only wondering why I always get this strange issue when I try to upload a font on my test instance...

**Reply by szaimen** [reply]
BTW, are the fonts by collabora correctly downloaded automatically to this location? `/var/lib/docker/volumes/nextcloud_aio_collabora_fonts/_data`

**Reply by tquest1** [reply]
> BTW, are the fonts by collabora correctly downloaded automatically to this location? `/var/lib/docker/volumes/nextcloud_aio_collabora_fonts/_data`

No, that directory is empty - they are uploaded to the richdocuments app folder which in my case (I have my data in a mounted volume) is: 
`/mnt/ncdata/appdata_oc5xzk00mgtl/richdocuments/fonts/`

I tried copying some fonts to that folder with the right permissions manually to see if they would show up, as uploading the fonts from the gui can only be done one at a time & that is very tedious if you have a lot of fonts. Is there a command that needs to be entered to 'scan' these fonts? I tried restarting the Collabora container & it doesn't work.

**Reply by szaimen** [reply]
The volume was introduced so that caching of fonts inside the container is possible. The question is now to which directory inside the containers they are uploaded to. Can you check inside the collabora container where they are stored?

**Reply by tquest1** [reply]
I've entered the Collabora container with: `docker exec --user root -it nextcloud-aio-collabora bash` & searched for any truetype fonts with `find / -name '*.ttf'` & I can only see the preinstalled fonts. Let me know if I'm going about this incorrectly - I'm far from an expert! 

The only place that I can see the uploaded fonts are in the previously mentioned: `/appdata_oc5xzk00mgtl/richdocuments/fonts/` in the Nextcloud data folder. I even did a search in the underlying Proxmox container volume where it is installed & the the results are the same.

**Reply by szaimen** [reply]
all right, then I might be wrong and they dont get downloaded into the collabora container even not temporarily.

cc @juliushaertl or @nextcloud/office do you know more? :)

**Comment by patrick-5546** [comment]
@szaimen were you able to resolve the `Font format not supported` error you were getting? I'm seeing it on my own instance now, which is running v4.4.1.

**Reply by szaimen** [reply]
Yes, see https://github.com/nextcloud/all-in-one/discussions/1859

**Comment by g-work** [comment]
Are there any news converning the second part of the question?

> Also - what is the correct way to remove some of the pre-installed fonts - I'd like to remove the unused 'Noto' font languages as the font drop down is too long with all of the various noto fonts.

I'm having the same issue since my users complained that the font list is way too long :)

**Reply by szaimen** [reply]
Sorry I dont know. Probably good to ask this at https://github.com/nextcloud/richdocuments/issues
